ndb/src/google/cloud/ndb/model.py

```diff
@@ -288,11 +287,32 @@ def __hash__(self):
         return hash((self.definition, self.state, self.id))
 
 
-class ModelAdapter:
-    __slots__ = ()
+def _entity_from_protobuf(protobuf):
+    """Deserialize an entity from a protobuffer.
 
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError
+    Args:
+        protobuf (google.cloud.datastore_v1.proto.entity.Entity): An
+            entity protobuf to be deserialized.
+
+    Returns:
+        .Model: The deserialized entity.
+    """
+    ds_entity = helpers.entity_from_protobuf(protobuf)
+    model_class = Model._lookup_model(ds_entity.kind)
+    entity = model_class()
```

Comment:
Do we want this? I.e. would calling `__new__` / bypassing `__init__` make more sense? What did the original code do (send a link?)?

Reply:
https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/model.py#L3194

Reply:
I have in other persistence systems used `model_class.__new__(model_class)` in this spot, so I'm fine with changing it. I was just blindly copying from the source implementation here.